pytorch/pytorch_edgeml/graph/rnn.py

```diff
@@ -1065,14 +1065,33 @@ def __init__(self, inputDim, outputDim, hiddenDim0, hiddenDim1, cellType):
         self.hiddenDim0 = hiddenDim0
         self.hiddenDim1 = hiddenDim1
         self.outputDim = outputDim
-        supportedCells = ['LSTM']
+        supportedCells = ['LSTM','FastGRNNCell','GRULRCell']
         assert cellType in supportedCells, 'Currently supported cells: %r' % supportedCells
         self.cellType = cellType
+        
         if self.cellType == 'LSTM':
             self.rnnClass = nn.LSTM
-
-        self.rnn0 = self.rnnClass(input_size=inputDim, hidden_size=hiddenDim0)
-        self.rnn1 = self.rnnClass(input_size=hiddenDim0, hidden_size=hiddenDim1)
+            self.rnn0 = self.rnnClass(input_size=inputDim, hidden_size=hiddenDim0)
+            self.rnn1 = self.rnnClass(input_size=hiddenDim0, hidden_size=hiddenDim1)
+        
+        if self.cellType == 'FastGRNNCell':
+            self.rnnClass = FastGRNN
+            self.rnn0 = self.rnnClass(input_size=inputDim, hidden_size=hiddenDim0, gate_non_linearity = fastCellArgs['gate_non_linearity'],
+                 update_non_linearity = fastCellArgs['update_non_linearity'], wRank = fastCellArgs['wRank'], uRank = fastCellArgs['uRank'],
+                 zetaInit = fastCellArgs['zetaInit'], nuInit = fastCellArgs['nuInit'], batch_first = fastCellArgs['batch_first'])
+            self.rnn1 = self.rnnClass(input_size=hiddenDim0, hidden_size=hiddenDim1, gate_non_linearity = fastCellArgs['gate_non_linearity'],
+                 update_non_linearity = fastCellArgs['update_non_linearity'], wRank = fastCellArgs['wRank'], uRank = fastCellArgs['uRank'],
+                 zetaInit = fastCellArgs['zetaInit'], nuInit = fastCellArgs['nuInit'], batch_first = fastCellArgs['batch_first'])
```

Comment:
You don't need any of these steps. Once you have made `cellArgs` a dictionary argument of the `__init__` function, you just have to do:

```
self.rnn0 = self.rnnClass(input_size=inputDim, hidden_size=hiddenDim0, **cellArgs)
```

Reply:
Resolved